src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseStringEqualsOverStringCompare.Fixer.cs

```diff
@@ -0,0 +1,218 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Operations;
+
+using Resx = Microsoft.NetCore.Analyzers.MicrosoftNetCoreAnalyzersResources;
+using RequiredSymbols = Microsoft.NetCore.Analyzers.Runtime.UseStringEqualsOverStringCompare.RequiredSymbols;
+
+namespace Microsoft.NetCore.Analyzers.Runtime
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic), Shared]
+    public sealed class UseStringEqualsOverStringCompareFixer : CodeFixProvider
+    {
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var document = context.Document;
+            var token = context.CancellationToken;
+            var semanticModel = await document.GetSemanticModelAsync(token).ConfigureAwait(false);
+
+            if (!RequiredSymbols.TryGetSymbols(semanticModel.Compilation, out var symbols))
+                return;
+
+            var root = await document.GetSyntaxRootAsync(token).ConfigureAwait(false);
+
+            if (semanticModel.GetOperation(root.FindNode(context.Span, getInnermostNodeForTie: true), token) is not IBinaryOperation violation)
+                return;
+
+            //  Get the replacer that applies to the reported violation. Bail out if no replacer 
+            //  matches the reported violation.
+            var replacer = GetOperationReplacers(symbols).FirstOrDefault(x => x.IsMatch(violation));
```

Comment:
Assert or throw here. The replacers are supposed to be strongly tied with the analyzer right?

Reply:
I'll replace `FirstOrDefault` with `First`.